### PR TITLE
gnrc_ipv6: accept packets for global dst at all interfaces

### PR DESCRIPTION
**Pull request by authmillenon**
Alternative to #4456.

### CODE REVIEW

**Comment by kaspar030** [comment]
How did you test this?

I tried this using the same setup as in #4456.
In that setup, the icmp echo is correctly received, but the reply is sent over the wrong interface.

This PR only touches forwarding (as only _pkt_not_for_me is changed).

**Comment by kaspar030** [comment]
> This PR only touches forwarding (as only _pkt_not_for_me is changed).

Sorry, not correct...

**Comment by kaspar030** [comment]
Ok, I tested a wrong setup (using link-local addresses and a fib default route).
With the same setup as in in #4456 (this time for real), I can ping the 6lowpan address of 6lbr.

File: sys/net/gnrc/network_layer/ipv6/gnrc_ipv6.c [hunk 1]
```diff
@@ -677,9 +677,14 @@ static inline bool _pkt_not_for_me(kernel_pid_t *iface, ipv6_hdr_t *hdr)
     if (ipv6_addr_is_loopback(&hdr->dst)) {
         return false;
     }
-    else if (*iface == KERNEL_PID_UNDEF) {
-        *iface = gnrc_ipv6_netif_find_by_addr(NULL, &hdr->dst);
-        return (*iface == KERNEL_PID_UNDEF);
+    else if ((!ipv6_addr_is_link_local(&hdr->dst)) ||
+             (*iface == KERNEL_PID_UNDEF)) {
+        kernel_pid_t if_pid = gnrc_ipv6_netif_find_by_addr(NULL, &hdr->dst);
+        if (*iface != KERNEL_PID_UNDEF) {
```
Comment (Yonezawa-T2) [inline sys/net/gnrc/network_layer/ipv6/gnrc_ipv6.c]: `*iface == KERNEL_PID_UNDEF`?

Comment (authmillenon) [inline sys/net/gnrc/network_layer/ipv6/gnrc_ipv6.c]: Who needs logic :P (yes)

Comment (authmillenon) [inline sys/net/gnrc/network_layer/ipv6/gnrc_ipv6.c]: Wait no, it should be changed in any case. :-)

Comment (authmillenon) [inline sys/net/gnrc/network_layer/ipv6/gnrc_ipv6.c]: okay, let's think about this ... to fix the @kaspar030's issue `*iface` must not be changed at all (in that case `*iface` shouldn't be `KERNEL_PID_UNDEF`). If no interface is given (`*iface == KERNEL_PID_UNDEF`) `*iface` should be overwritten. So your original comment was correct.

**Comment by authmillenon** [comment]
Addressed @Yonezawa-T2's comment.

**Comment by authmillenon** [comment]
@kaspar030 can you test again? Only read your corrections afterwards..... Now I'm not sure this is actually a fix. Will revert as soon as you say this isn't working anymore.

**Comment by cgundogan** [comment]
looks good. ACK

**Comment by OlegHahm** [comment]
Needs squashing.

**Comment by kaspar030** [comment]
I just re-checked that this fixes the bug where RIOT was not answering to all of it's addresses.

**Comment by authmillenon** [comment]
Squashed.

**Pull request merged**